common/rya.api/src/main/java/mvm/rya/api/instance/RyaDetailsToConfiguration.java

```diff
@@ -0,0 +1,86 @@
+package mvm.rya.api.instance;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+
+import mvm.rya.api.instance.RyaDetails.PCJIndexDetails;
+import mvm.rya.api.instance.RyaDetails.PCJIndexDetails.PCJDetails;
+
+/**
+ * Used to fetch {@link RyaDetails} from a {@link RyaDetailsRepository} and
+ * add them to the application's {@link Configuration}.
+ */
+public class RyaDetailsToConfiguration {
+    private static final Logger LOG = Logger.getLogger(RyaDetailsToConfiguration.class);
+    public static final String CONF_PCJ_IDS = "pcj.index.ids";
+    /**
+     * @param details - The {@link RyaDetails} to add to the {@link Configuration}.
+     * @param conf - The {@link Configuration} to add {@link RyaDetails} to.
+     */
+    public static void addRyaDetailsToConfiguration(final RyaDetails details, final Configuration conf) {
+        Preconditions.checkNotNull(details);
+        Preconditions.checkNotNull(conf);
+
+        checkAndSet(conf, ConfigurationFields.USE_ENTITY, details.getEntityCentricIndexDetails().isEnabled());
+        checkAndSet(conf, ConfigurationFields.USE_ENTITY, details.getEntityCentricIndexDetails().isEnabled());
+        checkAndSet(conf, ConfigurationFields.USE_FREETEXT, details.getFreeTextIndexDetails().isEnabled());
+        checkAndSet(conf, ConfigurationFields.USE_GEO, details.getGeoIndexDetails().isEnabled());
+        checkAndSet(conf, ConfigurationFields.USE_TEMPORAL, details.getTemporalIndexDetails().isEnabled());
+        final boolean pcjEnabled = details.getPCJIndexDetails().isEnabled();
+        checkAndSet(conf, ConfigurationFields.USE_PCJ, pcjEnabled);
+        if(pcjEnabled) {
+            final PCJIndexDetails pcjDetails = details.getPCJIndexDetails();
+            final List<String> pcjIds = new ArrayList<>();
+            for(final PCJDetails pcj : pcjDetails.getPCJDetails()) {
+                pcjIds.add(pcj.getId());
+            }
+            conf.setStrings(CONF_PCJ_IDS, pcjIds.toArray(new String[]{}));
```

Comment:
Do these values get used anywhere? I don't see any code updates that would indicate just the IDs are required instead of the entire table names.